### PR TITLE
Pyroscope: Change the grafana-pyroscope data source ID

### PR DESCRIPTION
**Pull request by aocenas**
Different approach to https://github.com/grafana/grafana/pull/67490.

Instead of just changing the ID of the data source this PR keeps both `phlare` and `grafana-pyroscope`. Both those data sources point to the same code base so it's effectively just an alias. To prevent showing both of them in the Add data source page and prevent too much random hardcoding, a new `hidden` property on plugin.json is added which prevents the plugin to be returned from `/api/plugins`.

This has the benefit of all the old data source instances to keep working without change, same for any provisioning files which still create `phlare` or any saved queries or URLs.

This still allows us to do a manual database and dashboard migration at some point without breaking anything on rollback.

### CODE REVIEW

File: pkg/plugins/plugins.go [hunk 1]
```diff
@@ -137,6 +137,9 @@ type JSONData struct {
 
 	// Backend (Datasource + Renderer + SecretsManager)
 	Executable string `json:"executable,omitempty"`
+
+	// Hides plugin from the plugins list but still allows data sources of that type to function
+	Hidden bool `json:"hidden,omitempty"`
```
Comment (aocenas) [inline pkg/plugins/plugins.go:142]: Seems like the plugindef.cue isn't yet used in the code?

**Comment by bergquist** [comment]
LGTM good from my POV 👍 

but I think @marefr or someone from plugin platform should review the hidden field as well.

**Comment by aocenas** [comment]
@marefr I don't have a strong opinion about the `hidden` attribute, could be done by filtering in the API. Thought this may be useful in the future and means removing the `phlare` datasource in the future would be easier.

**Review by wbrowne — COMMENTED** [review]
I think there's potentially a simpler way to solve this without adding a field to the plugin.json. I think adding `"builtIn": true` should give you what you need for the existing phlare datasource. Given this is a short term thing, I think this would be fine (see existing datasources `grafana`, `dashboard` and `mixed` which use this property).

I also think it would be better to go for `grafana-pyroscope-datasource` as that lines up better with out conventions.

**Comment by ryantxu** [comment]
alternate -- solution that makes this easier to do again (and supports panels!) https://github.com/grafana/grafana/pull/67867

**Comment by aocenas** [comment]
@wbrowne thanks for the tip will try that. I think though this doc comment should change then:
```
				// [internal only] Indicates whether the plugin is developed and shipped as part
				// of Grafana. Also known as a 'core plugin'.
```
I would assume all plugins in the Grafana repo are `core plugins` so this is a bit confusing.

@ryantxu looked at your PR and yeah it is closer to what I had in mind originally but wasn't sure how complex it would be. Also that said your PR does not seem to work for me. The old ID plugins get broken still.

> and supports panels!
Also this PR works for panels, so I am a bit confused now.

**Comment by ryantxu** [comment]
@aocenas -- can you try again?  I do not have a working phlare instance, so stopped at the "can i see it in a list" part 🤕 

re panels -- the alias logic will also work for panel plugins.  It is now hardcoded to only work for the "debug" panel, but this will allow us changing panel ids, not just datasources.

**Comment by aocenas** [comment]
Closing as this was solved here https://github.com/grafana/grafana/pull/67867

**Pull request closed without merging**